plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java

```diff
@@ -65,6 +65,7 @@
         implements Exchange
 {
     private static final Pattern PARTITION_FILE_NAME_PATTERN = Pattern.compile("(\\d+)_(\\d+)\\.data");
+    private static final Random random = new Random();
```

Comment:
`Random` is not thread safe. Prefer using `ThreadLocalRandom#current` inline.

Reply:
Thanks for pointing out. `Random` has become thread safe since Java 7.

But you are right that `ThreadLocalRandom` should be preferred here since it provides better performance.

Reply:
Oh, right. It's thread safe now. I didn't realize it.